app/Http/Controllers/MgtController.php

```diff
@@ -229,6 +229,29 @@ public function getControllerMentor($cid, $facility = null) {
             return redirect("/mgt/controller/$cid")->with("success", "Successfully removed mentor role");
         }
     }
+    
+    public function getControllerInstructor($cid, $facility) {
+        if (!RoleHelper::isVATUSAStaff() && !RoleHelper::isFacilitySeniorStaff()) {
+            return redirect('/mgt/controller/' . $cid)->with("error", "Access denied.");
+        }
+
+        $user = User::find($cid);
+        if (!$user) {
+            return redirect("/mgt/controller")->with("error", "User not found");
+        }
+
+        if ($facility == null) {
+            return redirect("/mgt/controller")->with("error", "Incorrect facility name");
+        }
+        
+        if($user->rating<8){
+            return redirect("/mgt/controller")->with("error", "User is not an Instructor");
+        }
```

Comment:
We should probably check RoleHelper::isInstructor() instead, as we only want to allow ADM and SUP if they're flagged as an instructor in their home facility. I believe a different function is used for the SUP INS version of this role. If not, we could add an exception if $facility is the user's home facility.